frontend/src/utils/game_utils.js

```diff
@@ -83,3 +83,53 @@ export const getSteps = (path, heightMap) => {
     }
     return steps;
 }
+
+const aboveGround = (x, y, z, heightMap) => {
+    const [xIdx, yIdx] = [round(x,0), round(y,0)];
+    if (xIdx < 0 || xIdx >= heightMap.length || yIdx < 0 || yIdx >= heightMap[0].length) return false;  
+    return z >= heightMap[xIdx][yIdx];
+}
+
+export const getStraightPath = (start, end, heightMap, speed=0.1) => {
```

Comment:
Can you add comment to this and `getParabolicPath()` explaining what they do and inputs/outputs.